airflow/cli/commands/connection_command.py

```diff
@@ -301,19 +301,26 @@ def connections_delete(args):
 def connections_import(args):
     """Imports connections from a file."""
     if os.path.exists(args.file):
-        _import_helper(args.file)
+        _import_helper(args.file, args.overwrite)
     else:
         raise SystemExit("Missing connections file.")
 
 
-def _import_helper(file_path):
-    """Load connections from a file and save them to the DB. On collision, skip."""
+def _import_helper(file_path, overwrite):
+    """
+    Load connections from a file and save them to the DB.
+    If `overwrite` is set to true, overwrite on collision.
+    """
     connections_dict = load_connections_dict(file_path)
     with create_session() as session:
         for conn_id, conn in connections_dict.items():
-            if session.query(Connection).filter(Connection.conn_id == conn_id).first():
+            conn_id_already_exists = session.query(Connection).filter(Connection.conn_id == conn_id).first()
+            if not overwrite and conn_id_already_exists:
                 print(f"Could not import connection {conn_id}: connection already exists.")
                 continue
+            if overwrite and conn_id_already_exists:
+                session.delete(conn_id_already_exists)
+                session.flush()
 
             session.add(conn)
```

Comment:
In the `overwrite=True` case I wonder if it’s a good idea to update the existing row instead of a delete-insert combination. Maybe even upsert on supported database backends?

Reply:
That's fair. I tried to get updates working, but ran into IntegrityErrors a bunch due to how flushes work. I fell back on delete/flush/create, but I'll take another look today to see if I can get update working.

Thanks for the review!

Reply:
In case this helps any future readers (or me 😁), the IntegrityErrors were due to the fact that I thought `conn_id` was the PK for connections. When I merged a new `Connection` into the session to overwrite an existing one with the same `conn_id`, the session wouldn't reconcile the existing and new entries.

While there is a unique constraint on `conn_id`, the PK is `id`. The two entries having different PKs with the same `conn_id` is what caused the IntegrityError.

I fixed that by setting `conn.id = existing_conn.id` and have upserts working.